tracer/src/Datadog.Trace.SourceGenerators/AspectsDefinitions/AspectsDefinitionsGenerator.cs

```diff
@@ -21,181 +22,216 @@
 [Generator]
 public class AspectsDefinitionsGenerator : IIncrementalGenerator
 {
+    private const string NullLiteral = "null";
+
     /// <inheritdoc />
     public void Initialize(IncrementalGeneratorInitializationContext context)
     {
-#if DEBUG__
-        if (!Debugger.IsAttached)
-        {
-            Debugger.Launch();
-        }
-#endif
-
-        // Register the attribute source
-
-        IncrementalValuesProvider<ClassDeclarationSyntax> classDeclarations =
-            context.SyntaxProvider.CreateSyntaxProvider(
-                        static (node, token) => IsAttributedClass(node, token),
-                        static (syntaxContext, token) => GetPotentialClassesForGeneration(syntaxContext, token))
-                   .Where(static m => m is not null)!;
-
-        IncrementalValuesProvider<AttributeData> assemblyAttributes =
-           context.CompilationProvider.SelectMany(static (compilation, _) => compilation.Assembly.GetAttributes());
-
-        IncrementalValueProvider<(Compilation Left, ImmutableArray<ClassDeclarationSyntax> Right)> compilationAndClasses =
-           context.CompilationProvider
-                  .Combine(classDeclarations.Collect());
-
-        IncrementalValueProvider<IReadOnlyList<string>> detailsToRender =
-           compilationAndClasses.Select(static (x, ct) => GetDefinitionsToWrite(x.Left, x.Right, ct));
-        context.RegisterSourceOutput(detailsToRender, static (spc, source) => Execute(source, spc));
+        // RegisterSource(context, "AspectAttribute");
+        // RegisterSource(context, "AspectClassAttribute");
+        // RegisterSource(context, "AspectCtorReplaceAttribute");
+        // RegisterSource(context, "AspectFilter");
+        // RegisterSource(context, "AspectMethodInsertAfterAttribute");
+        // RegisterSource(context, "AspectMethodInsertBeforeAttribute");
+        // RegisterSource(context, "AspectMethodReplaceAttribute");
+        // RegisterSource(context, "AspectType");
+        // RegisterSource(context, "VulnerabilityType");
+        // RegisterSource(context, "VulnerabilityTypeName");
+
+        var aspectsClassesToGenerate = context.SyntaxProvider
```

Comment:
nit: while I'm a big fan of `var` generally, I personally like to be specific about the type returned in these because it matters a _lot_ what they are for caching reasons 🙂

Reply:
Fixed